packages/vuetify/src/components/VOtpInput/VOtpInput.ts

```diff
@@ -0,0 +1,340 @@
+// Styles
+import '../VTextField/VTextField.sass'
+import './VOtpInput.sass'
+
+// Extensions
+import VInput from '../VInput'
+
+// Directives
+import ripple from '../../directives/ripple'
+
+// Utilities
+import { convertToUnit, keyCodes } from '../../util/helpers'
+import { breaking } from '../../util/console'
+
+// Types
+import mixins from '../../util/mixins'
+import { VNode } from 'vue'
+
+const baseMixins = mixins(
+  VInput,
+)
+interface options extends InstanceType<typeof baseMixins> {
+  $refs: {
+    input: HTMLInputElement[]
+  }
+}
+
+/* @vue/component */
+export default baseMixins.extend<options>().extend({
+  name: 'v-otp-input',
+
+  directives: {
+    ripple,
+  },
+
+  inheritAttrs: false,
+
+  props: {
+    length: {
+      type: Number,
+      default: 6,
+    },
+    type: {
+      type: String,
+      default: 'text',
+    },
+  },
+
+  data: () => ({
+    badInput: false,
+    initialValue: null,
+    isBooted: false,
+    otp: [] as string[],
+  }),
+
+  computed: {
+    classes (): object {
+      return {
+        ...VInput.options.computed.classes.call(this),
+        'v-input--hide-details': true,
+        'v-text-field': true,
+        'v-text-field--enclosed': true,
+        'v-text-field--outlined': true,
```

Comment:
This relies on us not changing any class names. Can we pass down styling props to v-text-field if it's going to be used?